packages/amplify-provider-awscloudformation/src/zip-util.js

```diff
@@ -1,56 +0,0 @@
-const extract = require('extract-zip');
-const fs = require('fs-extra');
-const path = require('path');
-const { fileLogger } = require('./utils/aws-logger');
-const logger = fileLogger('zip-util');
-
-function downloadZip(s3, tempDir, zipFileName, envName) {
-  return new Promise((resolve, reject) => {
-    const log = logger('downloadZip.s3.getFile', [{ Key: zipFileName }, envName]);
-    log();
-    s3.getFile(
-      {
-        Key: zipFileName,
-      },
-      envName,
-    )
-      .then((objectResult, objectError) => {
-        if (objectError) {
-          reject(objectError);
-          return;
-        }
-
-        fs.ensureDirSync(tempDir);
-
-        const buff = Buffer.from(objectResult);
-        const tempfile = `${tempDir}/${zipFileName}`;
-
-        fs.writeFile(tempfile, buff, err => {
```

Comment:
the fix for flaky delete in notifications-sms-pull.test.ts as sometimes the writeFile was not finishing before the extractZip method was called in https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-provider-awscloudformation/src/delete-env.ts#L58-L59

Reply:
Nice find!